src/main/scala/cromwell/engine/backend/jes/JesAttributes.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.engine.backend.jes
+
+import java.net.URL
+
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.util.ConfigUtil._
+
+case class JesAttributes(applicationName: String,
+                         project: String,
+                         executionBucket: String,
+                         endpointUrl: URL,
+                         authMode: GcsAuthMode,
+                         docker: Option[DockerAuthInformation])
+object JesAttributes {
+
+  def apply(): JesAttributes = {
+    val jesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+    val refConf: Config = ConfigFactory.parseResources("jes.conf")
+
+    jesConf.checkValidWrapped(refConf, "Jes")
+    jesConf.warnNotRecognized(refConf, "Jes", "dockerAccount", "dockerToken")
```

Comment:
also the reference configuration does not have to be in a file, we could make it a hardcoded string or a `Map` or pretty much anything really since `ConfigFactory` can parse a lot of things.
I created a separate `jes.conf` file but I'm not convinced it's the best choice.